Python/bytecodes.c

```diff
@@ -190,7 +190,24 @@ dummy_func(
             }
         }
 
+        op(_LOAD_BYTECODE, (--)) {
+            #ifdef Py_GIL_DISABLED
+            _Py_CODEUNIT *bytecode = _PyEval_GetExecutableCode(_PyFrame_GetCode(frame));
+            ERROR_IF(bytecode == NULL, error);
+            if (frame->bytecode != bytecode) {
+                int off = this_instr - frame->bytecode;
+                frame->bytecode = bytecode;
+                frame->instr_ptr = frame->bytecode + off;
+                this_instr = frame->instr_ptr;
```

Comment:
We need to do this so that `this_instr` is correct in any uops that follow. Concretely, in `_MAYBE_INSTRUMENT`.

Reply:
This is not how it works, `this_instr` is a pointer to the current instruction. You can read it, but not write it.
The code generator is free to create and rewrite it as it wants.

If you need to change `next_instr` for following code it will need to be its own instruction, like a jump.

Reply:
> This is not how it works, this_instr is a pointer to the current instruction.

I understand that, but `this_instr` will be incorrect after we reload the bytecode. There may be uops that follow `_LOAD_BYTECODE` that use `this_instr` (`_MAYBE_INSTRUMENT`) and we want them to use the correct value. 

IIRC, right now the code generator sets `this_instr` at the beginning of the generated opcode body. I could modify the code generator to recognize `_LOAD_BYTECODE` and generate code to reload `this_instr`  after it, if necessary (i.e. if any of the following uops use it). I don't like the special casing, but maybe that's slightly better than what's here. What do you think?

Reply:
I ended up working around this by dispatching to the same instruction when we need to reload the bytecode. It'll make the first function call a tad bit slower, but we can figure out a better solution if needed.